### ISSUE TITLE
Inconsistent behavior on run()

### ISSUE THREAD

**Issue by waghanza** [dom96/httpbeast#57]
Hi @dom96,

When specifying settings (port, bind address, ...), the domain is set to `AF_UNSPEC`. However, without settings, the domain is set to `AF_INET` 

This make
https://github.com/dom96/httpbeast/blob/b33dba93a134faa2adc3b991cb69890375987076/src/httpbeast.nim#L464
and 
https://github.com/dom96/httpbeast/blob/b33dba93a134faa2adc3b991cb69890375987076/src/httpbeast.nim#L492
have different behavior (except if we specify `domain` for the first one)

Regards,

----------
Related to https://github.com/the-benchmarker/web-frameworks/pull/4554

**Comment by dom96** [comment]
Will keep this to remind me to add an error when `Settings` is initialised incorrectly.

**Comment by waghanza** [comment]
How can you override `Settings` @dom96 ?

**Comment by dom96** [comment]
https://github.com/dom96/httpbeast/blob/master/src/httpbeast.nim#L557

**Comment by waghanza** [comment]
exactly, but seems that you need to set by hand in https://github.com/mattaylor/whip/blob/a9fd1f04ba0167727dd0dbf9235791f708147992/whip.nim#L132 @mattaylor

@dom96 maybe a good idea to allow it to be overwritten by end-user (I mean with a compilation option or nim.cfg)
